src/crawlee/storage_clients/_file_system/_request_queue_client.py

```diff
@@ -248,13 +230,40 @@ async def open(
                     pending_request_count=0,
                     total_request_count=0,
                 )
-                client = cls(
-                    metadata=metadata,
-                    storage_dir=storage_dir,
-                    lock=asyncio.Lock(),
-                )
-                await client._state.initialize()
-                await client._update_metadata()
+                client = await cls._create_client(metadata=metadata, storage_dir=storage_dir)
+
+        return client
+
+    @classmethod
+    async def _create_client(
+        cls, metadata: RequestQueueMetadata, storage_dir: Path, *, update_accessed_at: bool = False
+    ) -> FileSystemRequestQueueClient:
+        """Create client from metadata and storage directory."""
+        from crawlee.storage_clients import FileSystemStorageClient  # noqa: PLC0415 avoid circular imports
+        from crawlee.storages._key_value_store import KeyValueStore  # noqa: PLC0415 avoid circular imports
+
+        # Prepare kvs for recoverable state
+        kvs_client = await FileSystemStorageClient().create_kvs_client(name=f'__RQ_STATE_{metadata.id}')
```

Comment:
Wait a minute. This is a subtle bug waiting to happen - the `FileSystemStorageClient` that creates this RQ client might have a different configuration than the one you create here.

Can you pass the actual storage client down to this method instead?